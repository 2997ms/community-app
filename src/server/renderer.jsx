/**
 * This module implements ExpressJS middleware for server-side rendering of
 * the App.
 */

import config from 'config';
import React from 'react';
import ReactDOM from 'react-dom/server';
import { StaticRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import serializeJs from 'serialize-javascript';
import { isDev } from 'utils/isomorphy';

import App from '../shared';

/* This is always initial state of the store. Later we'll have to provide a way
 * to put the store into correct state depending on the demanded route. */
import storeFactory from '../shared/store-factory';

export default (req, res) => {
  storeFactory(req).then((store) => {
    const context = {};
    const appHtml = ReactDOM.renderToString((
      <Provider store={store}>
        <StaticRouter
          context={context}
          location={req.url}
        >
          <App />
        </StaticRouter>
      </Provider>
    ));
    if (context.status) res.status(context.status);
    res.send((
      `<!DOCTYPE html>
      <html>
        <head>
          <title>Topcoder</title>
          ${
            /* In dev environment styles are embed into JS bundle,
              and we should not include this link into the page, otherwise
              the page will initially load production /style.css and it may
              result in some wierd behavior while loading. */
            isDev() ? '' : '<link rel="stylesheet" href="/style.css" />'
          }
        </head>
        <body>
          <div id="react-view">${appHtml}</div>
          <script type="application/javascript">
            window.CONFIG = ${serializeJs(config, { isJSON: true })}
            window.ISTATE = ${serializeJs(store.getState(), { isJSON: true })}
          </script>
          <script type="application/javascript" src="/bundle.js"></script>
        </body>
      </html>`
    ));
  });
};
