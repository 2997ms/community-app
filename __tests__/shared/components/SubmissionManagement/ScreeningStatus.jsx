import React from 'react';
import renderer from 'react-test-renderer';

import ScreeningStatus from 'components/SubmissionManagement/ScreeningStatus';
import mockScreeningObject from './__mocks__/screening-object.json';

test('renders correctly', () => {
  const html = renderer.create((
    <ScreeningStatus
      screeningObject={mockScreeningObject}
      submissionId="12345"
    />
  )).toJSON();
  expect(html).toMatchSnapshot();
});
