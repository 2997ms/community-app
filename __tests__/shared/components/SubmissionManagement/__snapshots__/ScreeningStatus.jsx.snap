// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  onClick={[Function]}
>
  <div
    className="src-shared-components-SubmissionManagement-ScreeningStatus-___styles__screening-status___JV4BF src-shared-components-SubmissionManagement-ScreeningStatus-___styles__has-warn___8jqPg"
  >
    <span
      className="src-shared-components-SubmissionManagement-ScreeningStatus-___styles__status___1CCjJ"
    >
      
    </span>
    <span
      className="src-shared-components-SubmissionManagement-ScreeningStatus-___styles__warning___6tSHD"
    >
      2 
       warnings
    </span>
  </div>
</button>
`;
